diesel/src/pg/expression/functions.rs

```diff
@@ -748,3 +748,36 @@ define_sql_function! {
     /// ```
     fn array_append<Arr: ArrayOrNullableArray<Inner=T> + SingleValue, T: SingleValue>(a: Arr, e: T) -> Array<T>;
 }
+
+#[cfg(feature = "postgres_backend")]
+define_sql_function! {
+    /// Returns the number of dimensions of the array
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::array_ndims;
+    /// #     use diesel::sql_types::{Nullable, Array, Integer};
+    /// #     let connection = &mut establish_connection();
+    ///
+    /// let dims = diesel::select(array_ndims::<Array<_>, _>(vec![1, 2]))
```

Comment:
Not sure why, but it fails to infer the types for `Array` here.

Reply:
It works on `array_append` because array append has a second argument which is defined
```suggestion
    /// let dims = diesel::select(array_ndims::<Array<Integer>, _>(vec![1, 2]))
```